### PR TITLE
Add missing properties relevant to listing all invoices

### LINKED ISSUE #242: Mising properties on account's invoices
When fetching an account's invoices it's seems there are some missing properties:

- `line_items -> adjustments -> proration_rate`
- `transactions -> transaction -> ip_address`
- `transactions -> transaction -> approval_code`
- `transactions -> transaction -> failure_type`
- `transactions -> transaction -> gateway_error_codes`
- `transactions -> transaction -> cvv_result`

Also there seems that` transactions -> transaction -> details -> account -> billing_info -> vat_number` comes with a different name from the Recurly's API.

Related to: [Missing address and shipping address in invoice #240](https://github.com/killbilling/recurly-java-library/issues/240)

### PR DESCRIPTION
**Pull request by aaron-suarez**
Fixes #242 

Playground script:

```java
package com.recurly.testrig;

import com.ning.billing.recurly.RecurlyClient;
import com.ning.billing.recurly.model.Invoices;
import com.ning.billing.recurly.model.Invoice;
import com.ning.billing.recurly.model.Adjustment;
import com.ning.billing.recurly.model.Transaction;

public class ListInvoices {
    public static void run(final RecurlyClient client) {
        try {
            client.open();
            final Invoices invoices = client.getInvoices();

            for (Invoice inv: invoices) {
                for (Adjustment adj: inv.getLineItems()) {
                    System.out.println("Refundable Total: " + adj.getRefundableTotalInCents());
                    System.out.println("State: " + adj.getState());
                    System.out.println("Proration Rate: " + adj.getProrationRate());
                }

                for (Transaction trans: inv.getTransactions()) {
                    System.out.println("IP Address: " + trans.getIpAddress());
                    System.out.println("Approval Code: " + trans.getApprovalCode());
                    System.out.println("Failure Type: " + trans.getFailureType());
                    if (trans.getCvvResult() != null)
                    {
                        System.out.println("CVV Result: " + trans.getCvvResult().getCode() + " " + trans.getCvvResult().getMessage());
                    }
                    System.out.println("Gateway Error Codes: " + trans.getGatewayErrorCodes());
                    System.out.println("Payment Method: " + trans.getPaymentMethod());
                    System.out.println("Message: " + trans.getMessage());
                    System.out.println("Company: " + trans.getDetails().getAccount().getCompanyName());
                }
            }
        } catch (Exception e) {
            e.printStackTrace();
        } finally {
          client.close();
        }
    }
}
```

### CODE REVIEW

File: src/main/java/com/ning/billing/recurly/model/AbstractTransaction.java [hunk 1]
```diff
@@ -286,6 +317,12 @@ public boolean equals(final Object o) {
...
             return false;
         }
+        if (paymentMethod != null ? !paymentMethod.equals(that.paymentMethod) : that.paymentMethod != null) {
+            return false;
+        }
+        if (message != null ? !message.equals(that.message) : that.message != null) {
+            return false;
+        }
 
         return true;
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/AbstractTransaction.java]: Missing `gatewayErrorCodes`?

File: src/main/java/com/ning/billing/recurly/model/AbstractTransaction.java [hunk 2]
```diff
@@ -303,6 +340,7 @@ public int hashCode() {
                 transactionError,
                 source,
                 ipAddress,
+                message,
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/AbstractTransaction.java]:
```suggestion
                message,
                gatewayErrorCodes,
                paymentMethod,
```

File: src/main/java/com/ning/billing/recurly/model/Account.java [hunk 3]
```diff
@@ -402,7 +412,7 @@ public String toString() {
         sb.append(", ccEmails='").append(ccEmails).append('\'');
         sb.append(", firstName='").append(firstName).append('\'');
         sb.append(", lastName='").append(lastName).append('\'');
-        sb.append(", companyName='").append(companyName).append('\'');
+        sb.append(", companyName='").append(this.getCompanyName()).append('\'');
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/Account.java]: Might want to update `equals`/`hashCode` as well?

File: src/main/java/com/ning/billing/recurly/model/AbstractTransaction.java [hunk 4]
```diff
@@ -234,6 +243,28 @@ public void setAvsResultPostal(final Object avsResultPostal) {
...
+
+    public String getPaymentMethod() {
+        return paymentMethod;
+    }
+
+    public void setPaymentMethod(final Object paymentMethod) {
+        this.paymentMethod = stringOrNull(paymentMethod);
+    }
+
+    public String getMessage() { return this.message; }
```
Comment (pierre) [inline src/main/java/com/ning/billing/recurly/model/AbstractTransaction.java]: Nit: style.

**Comment by aaron-suarez** [comment]
Amended the commit to incorporate Pierre's feedback

**Pull request merged**